### ISSUE TITLE
--dir/--pkg is deprecated

### ISSUE THREAD

**Issue by DOG248** [Almamu/linux-wallpaperengine#55]
I always get this error when try to load a wallpaper:
```
--dir/--pkg is deprecated and not used anymore
terminate called after throwing an instance of 'std::runtime_error'
  what():  Cannot find the file in any of the containers
[1]    6533 abort  
```

Distro: Void Linux

**Comment by Crylia** [comment]
I got the same problem 
![grafik](https://user-images.githubusercontent.com/50151522/140384764-a874315d-9986-4414-9f95-2a13eb2f18d3.png)
Workshop id is in the link.
I tried both --pkg and --dir but both give this error

**Comment by bonn2** [comment]
Also having the same error except it is also giving a BadMatch error
```
$ ./wallengine --pkg /home/brett/.steam/steam/steamapps/workshop/content/431960/2148815645/ --screen-root HDMI-0 --fps 30
--dir/--pkg is deprecated and not used anymore
Found requested screen: HDMI-0 -> 0x180:1440x900
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  152 (GLX)
  Minor opcode of failed request:  5 (X_GLXMakeCurrent)
  Serial number of failed request:  149
  Current serial number in output stream:  149```

**Comment by Crylia** [comment]
I may want to add that even using the exact Wallpaper you used in the readme wont work no matter what I try.

**Comment by Almamu** [comment]
> I got the same problem ![grafik](https://user-images.githubusercontent.com/50151522/140384764-a874315d-9986-4414-9f95-2a13eb2f18d3.png) Workshop id is in the link. I tried both --pkg and --dir but both give this error

Can you show me the contents of that folder? The error message you're getting sounds like one of the files that the background requires cannot be found.



> 
> 
> Also having the same error except it is also giving a BadMatch error
> 
> ```
> $ ./wallengine --pkg /home/brett/.steam/steam/steamapps/workshop/content/431960/2148815645/ --screen-root HDMI-0 --fps 30
> --dir/--pkg is deprecated and not used anymore
> Found requested screen: HDMI-0 -> 0x180:1440x900
> X Error of failed request:  BadMatch (invalid parameter attributes)
>   Major opcode of failed request:  152 (GLX)
>   Minor opcode of failed request:  5 (X_GLXMakeCurrent)
>   Serial number of failed request:  149
>   Current serial number in output stream:  149```
> ```

Your error is different, what GPU are you running and what drivers?

**Comment by Essasetic** [comment]
Came to report the same issue:
`--dir/--pkg is deprecated and not used anymore
terminate called after throwing an instance of 'std::runtime_error'
  what():  Projection must have width
Aborted (core dumped)`

Distro: Arch Linux
Wallpaper: [https://steamcommunity.com/sharedfiles/filedetails/?id=1979606285](url)

**Comment by Almamu** [comment]
> 
> 
> Came to report the same issue: `--dir/--pkg is deprecated and not used anymore terminate called after throwing an instance of 'std::runtime_error' what(): Projection must have width Aborted (core dumped)`
> 
> ![image](https://user-images.githubusercontent.com/34970796/141024832-9b12955e-3a3d-423f-9877-9fdfa9990fe1.png)
> 
> Distro: Arch Linux Wallpaper: [https://steamcommunity.com/sharedfiles/filedetails/?id=1979606285](url)

In this case the background you're trying to display uses 3D models, which are not supported yet, so the app just crashes because the camera information is different.

**Comment by Crylia** [comment]
![grafik](https://user-images.githubusercontent.com/50151522/141104454-9c69707b-fbe1-4548-b92d-480cf238710e.png)
I'm using a Nvidia GTX 1060 6gb and on my laptop a GTX 1650 Max-Q.
Both with the 470.82 Driver

**Comment by n00b-spectre** [comment]
None of the backgrounds are working for me. I've tried the following one:
- [Summer Time](https://steamcommunity.com/sharedfiles/filedetails/?id=2477602742)
- [Cyberpunk San José 2077](https://steamcommunity.com/sharedfiles/filedetails/?id=2132578845)
- [Late Night Neon Idle Animation](https://steamcommunity.com/sharedfiles/filedetails/?id=1184945377)
- [Autumn](https://steamcommunity.com/sharedfiles/filedetails/?id=2133256742)
- [Tokyo Street with Day Night Cycle and Clock](https://steamcommunity.com/sharedfiles/filedetails/?id=1685364754)
- [Lofi Cafe](https://steamcommunity.com/sharedfiles/filedetails/?id=2370927443)
- [Serenity - Anime Landscape](https://steamcommunity.com/sharedfiles/filedetails/?id=1132505365)

The last one showed me some new error though!

![New Error](https://images2.imgbox.com/0f/94/iydUjiec_o.png)

**Comment by mcfarlandry** [comment]
I am also having the same issue.

![wallpaperengineerror](https://user-images.githubusercontent.com/65938832/143730866-653928a1-48ae-4935-8318-172036c0a742.png)

I am using the wallpaper [𝒮𝓎𝓂𝓅𝒽𝑜𝓃𝓎 ♪](https://steamcommunity.com/sharedfiles/filedetails/?id=2618897643). I put the wallpaper right in the directory with the binary.

**Comment by Almamu** [comment]
I've uploaded some fixes that should get rid of most of the shader errors (WE updated default assets and added some stuff that didn't really fit well with the current shader compiler), should be fixed since 9d2e9b43ee2eeb772ee65c0c9a8d72018ad62887.

I've also fixed some issues loading assets. This should get more backgrounds to display. I'll look into the ones that still have issues and try to address them in the short term so more and more backgrounds load properly.
